lib/gateway-kernel.coffee

```diff
@@ -0,0 +1,94 @@
+child_process = require 'child_process'
+path = require 'path'
+
+_ = require 'lodash'
+uuid = require 'uuid'
+services = require('./jupyter-js-services-shim')
+
+Kernel = require './kernel'
+
+class GatewayKernel extends Kernel
+    constructor: (kernelSpec, session) ->
+        super kernelSpec
+
+        @session = session
+        @session.statusChanged.connect(@_onStatusChange, this)
+        @_onStatusChange() # Set initial status correctly
+
+    interrupt: ->
+        @session.kernel.interrupt()
+
+    restart: ->
+        @session.kernel.restart()
+
+    _onStatusChange: ->
+        @statusView.setStatus @session.status
+
+    _onIOPub: (message, onResults) ->
+        msg_type = message.header.msg_type
+
+        # TODO(nikita): implement support for watches
+
+        if onResults?
+            console.log('got msg')
+            console.log(message)
+            result = @_parseIOMessage(message)
+            if result?
+                onResults result
+
+
+    execute: (code, onResults) ->
```

Comment:
@rgbkrk Kyle, is this correct? Does `jupyter-js-services` remove the user's ability to set `msg_id`? Isn't `msg_id` part of the [Jupyter Messaging Protocol](http://jupyter-client.readthedocs.io/en/latest/messaging.html#general-message-format)?